### PR TITLE
added cancellation token to json rpc tracers

### PR DESCRIPTION
**Pull request by SebastianDremo**

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugModuleFactory.cs [hunk 1]
```diff
@@ -72,7 +76,9 @@ public override IDebugModule Create()
             ReadOnlyBlockTree readOnlyTree = new ReadOnlyBlockTree(_blockTree);
             ReadOnlyTxProcessingEnv txEnv = new ReadOnlyTxProcessingEnv(readOnlyDbProvider, readOnlyTree, _specProvider, _logManager);
             ReadOnlyChainProcessingEnv readOnlyChainProcessingEnv = new ReadOnlyChainProcessingEnv(txEnv, _blockValidator, _recoveryStep, _rewardCalculatorSource.Get(txEnv.TransactionProcessor), _receiptStorage, readOnlyDbProvider, _specProvider, _logManager);
-            IGethStyleTracer tracer = new GethStyleTracer(readOnlyChainProcessingEnv.ChainProcessor, _receiptStorage, new ReadOnlyBlockTree(_blockTree));
+            CancellationToken cancellationToken = new CancellationTokenSource(TimeSpan.FromSeconds(5)).Token;
```
Comment (SebastianDremo) [inline src/Nethermind/Nethermind.JsonRpc/Modules/DebugModule/DebugModuleFactory.cs:79]: get TracertTimeout from config

File: src/Nethermind/Nethermind.JsonRpc/ErrorType.cs [hunk 2]
```diff
@@ -89,5 +89,10 @@ public static class ErrorCodes
         /// 
         /// </summary>
         public const int ExecutionError = -32015;
+         
+        /// <summary>
+        /// Request exceedes defined tracer timeout limit
+        /// </sumary>
+        public const int TracerTimeout = -32016;
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.JsonRpc/ErrorType.cs:96]: what is the number thrown by Geth?

File: src/Nethermind/Nethermind.JsonRpc/IJsonRpcConfig.cs [hunk 3]
```diff
@@ -26,6 +26,12 @@ public interface IJsonRpcConfig : IConfig
         [ConfigItem(Description = "Host for JSON RPC calls. Ensure the firewall is configured when enabling JSON RPC. If it does not work with 117.0.0.1 try something like 10.0.0.4 or 192.168.0.1", DefaultValue = "\"127.0.0.1\"")]
         string Host { get; set; }
 
+        [ConfigItem(Description = "Defines whether the JSON RPC tracers will have timeout enabled. Default timeout is 5 seconds per tracer usage.", DefaultValue = "false")]
+        bool TracerTimeoutEnabled { get; set; }
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.JsonRpc/IJsonRpcConfig.cs:30]: remove this and always enable tracer, just set the default timeout to something higher (20 seconds)

File: src/Nethermind/Nethermind.JsonRpc/IJsonRpcConfig.cs [hunk 4]
```diff
@@ -26,6 +26,12 @@ public interface IJsonRpcConfig : IConfig
...
+
+        [ConfigItem(Description = "JSON RPC tracers' timeout value given in seconds.", DefaultValue = "5")] 
+        int TracerTimeout { get; set; }
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.JsonRpc/IJsonRpcConfig.cs:33]: in milliseconds,, not seconds

File: src/Nethermind/Nethermind.Runner/configs/mainnet.cfg [hunk 5]
```diff
@@ -17,6 +17,8 @@
   },
   "JsonRpc": {
     "Enabled": false,
+    "TracerTimeout": 5,
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Runner/configs/mainnet.cfg:20]: ConfigFileTests should be added

File: src/Nethermind/Nethermind.JsonRpc/ErrorType.cs (hunk 2 above)
Comment (SebastianDremo) [reply src/Nethermind/Nethermind.JsonRpc/ErrorType.cs:96]: Geth doesn't have JsonRpc error code for timeout 
[Geth's jsonRpc errors](https://github.com/ethereum/go-ethereum/blob/beabf95ad7fd80db83944400348c1e42ac3b8c35/rpc/errors.go#L34)

it have timeouts for http client so I guess they just throw 408 in response 
[Geth's HttpTimeout](https://github.com/ethereum/go-ethereum/blob/master/rpc/http.go#L71)

**Pull request merged**